ios/Exponent/Kernel/Views/EXAppViewController.m

```diff
@@ -315,6 +318,34 @@ - (void)errorViewDidSelectRetry:(EXErrorView *)errorView
 
 - (UIInterfaceOrientationMask)supportedInterfaceOrientations
 {
+  NSString *validatedVersion = [[EXVersions sharedInstance] availableSdkVersionForManifest:_appRecord.appLoader.manifest];
+  NSString *majorSDKVersion = [validatedVersion componentsSeparatedByString:@"."][0];
+  if ([validatedVersion length] != 0 && [majorSDKVersion integerValue] <= 35) {
+    return [self legacySupportedInterfaceOrientations];
+  }
+  
+  if (_appRecord.experienceId == nil) {
+    return UIInterfaceOrientationMaskAllButUpsideDown;
+  }
+  EXScreenOrientationRegistry *registry = (EXScreenOrientationRegistry *)[UMModuleRegistryProvider getSingletonModuleForClass:[EXScreenOrientationRegistry class]];
+  if ([registry doesKeyExistForAppId:_appRecord.experienceId]) {
+    return [registry orientationMaskForAppId:_appRecord.experienceId];
+  } else if (_appRecord.appLoader.manifest) {
+    NSString *orientationConfig = _appRecord.appLoader.manifest[@"orientation"];
```

Comment:
This logic is almost exactly the same as `legacySupportedInterfaceOrientations`, right? Couldn't we merge them?